go.mod

```diff
@@ -22,6 +22,7 @@ require (
 	github.com/hashicorp/go-cleanhttp v0.5.1 // indirect
 	github.com/liamg/jfather v0.0.7 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
+	github.com/whilp/git-urls v1.0.0 // indirect
```

Comment:
Not sure how you'll feel about adding this little package. But still seemed better than manually parsing, especially if more git-based parsers are added later.